explore/src/main/scala/explore/constraints/ConstraintsQueries.scala

```diff
@@ -3,110 +3,135 @@
 
 package explore.constraints
 
-import cats.effect.ContextShift
+import cats.Endo
 import cats.effect.IO
 import clue.GraphQLOperation
 import clue.data.syntax._
 import clue.macros.GraphQL
-import crystal.ViewF
-import explore.GraphQLSchemas.ExploreDB.Types._
+import eu.timepit.refined.types.string.NonEmptyString
 import explore.GraphQLSchemas._
-import explore.components.graphql.SubscriptionRenderMod
+import explore.GraphQLSchemas.ObservationDB.Types._
 import explore.implicits._
-import explore.model.Constraints
-import explore.model.decoders._
-import explore.model.enum.CloudCover
-import explore.model.enum.ImageQuality
-import explore.model.enum.SkyBackground
-import explore.model.enum.WaterVapor
+import explore.model.ConstraintSetModel
+import explore.model.ElevationRange
 import explore.model.reusability._
+import explore.undo.UndoableView
 import explore.undo.Undoer
-import japgolly.scalajs.react.vdom.html_<^._
+import lucuma.core.model.ConstraintSet
+import lucuma.core.enum._
+import lucuma.ui.reusability._
 import monocle.Lens
-import monocle.macros.Lenses
+import explore.model.AirMassRange
+import explore.model.HourAngleRange
 
 object ConstraintsQueries {
   @GraphQL
-  object Subscription extends GraphQLOperation[ExploreDB] {
+  object ConstraintSetQuery extends GraphQLOperation[ObservationDB] {
     val document = """
-      subscription ($id: uuid!) {
-        constraints(where: {id: {_eq: $id}}) {
+      query($id: ConstraintSetId!) {
+        constraintSet(constraintSetId: $id) {
           id
           name
-          cloud_cover
-          image_quality
-          sky_background
-          water_vapor
+          cloudExtinction
+          imageQuality
+          skyBackground
+          waterVapor
+          elevationRange {
+            type: __typename
```

Comment:
What's the type used for?

Reply:
It is needed in the decoder to determine whether to decode an AirMassRange or an HourAngleRange.